io-engine/src/bdev/nexus/nexus_bdev_rebuild.rs

```diff
@@ -418,6 +430,14 @@ impl<'n> Nexus<'n> {
         }
 
         self.reconfigure(DrEvent::ChildRebuild).await;
+        // Translate the job into a new rebuild record and push into history.
+        trace!(
+            "Create Rebuild Record for child {} of nexus {}",
+            job.dst_uri,
+            job.nexus_name
+        );
+        let hist = unsafe { &mut self.get_unchecked_mut().rebuild_history };
```

Comment:
why is this needed?

Reply:
That's because of how Pin<> works. You cannot that easily get write access to fields of a pinned object. If you could, you could easily moved out values from it. So a pinned object can only be written in an unsafe block, thus showing a potential place violating Pin<>'s restrictions.